jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientTLSTest.java

```diff
@@ -1012,18 +1013,40 @@ public void testForcedNonDomainSNI() throws Exception
             .scheme(HttpScheme.HTTPS.asString())
             .send();
         assertEquals(HttpStatus.OK_200, response2.getStatus());
+    }
 
-        /* TODO Fix. See #6624
-        if (Net.isIpv6InterfaceAvailable())
+    @Test
+    @EnabledForJreRange(max = JRE.JAVA_16, disabledReason = "Since Java 17, SNI host names can only have letter|digit|hyphen characters.")
```

Comment:
This does not feel right.  We are relying on implementation knowledge that an arg check is left out of one version of a method in java < 17.  There is no reason that this arg check might not be added to a dot release of java 8-16.